drivers/spi/spi_esp32_spim.c

```diff
@@ -216,16 +216,33 @@ static int spi_esp32_init(const struct device *dev)
 	int err;
 	const struct spi_esp32_config *cfg = dev->config;
 	struct spi_esp32_data *data = dev->data;
+	spi_hal_context_t *hal = &data->hal;
 
 	if (!cfg->clock_dev) {
 		return -EINVAL;
 	}
 
+	if (!device_is_ready(cfg->clock_dev)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	/* enables SPI peripheral */
+	if (clock_control_on(cfg->clock_dev, cfg->clock_subsys)) {
```

Comment:
This error is misleading IMO. What is the reason for returning `EIO` error when the peripheral clocks are already enabled after the CPU reset?